src/components/include/utils/threads/thread.h

```diff
@@ -169,15 +192,20 @@ class Thread {
    * be used as if it were newly constructed (i.e., Start may be called again).
    *
    * Stop may be called multiple times and is simply ignored if the thread is
-   * already stopped.
+   * already stopped. Thread safe.
    */
   void stop();
 
   /**
    * @brief Starts join procedure.
-   * @param[in] force_stop If true stop() is called first otherwise is skipped.
+   * @param[in] force_stop for controlling call of stop() method first.
+   * If it is true stop() is called and stopped_ flag is set to true in order
+   * to command/force the thread to stop.
+   * If it is false stop() method is not called and stopped_ flag is not set.
+   * It is up to the thread delegate when/how will end.
+   * In both cases join() waits thread delegate to finish.
    */
-  void join(const bool force_stop = true);
+  void join(const bool force_stop);
```

Comment:
@orhan-mehmedov I have one general question... Do we really need such interface change (I mean adding `force_stop` parameter here)? Are we going to use `join` somewhere with this parameter `false` value?